### PR TITLE
Add tutorial “import pytorch models”

### PR DESCRIPTION
**Pull request by marigoold**
此PR完成了：
介绍如何将PyTorch的模型结构嵌入到Libai中使用，使用了MobileNetV2举例。

### CODE REVIEW

File: docs/source/tutorials/Import Pytorch Models.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Import PyTorch models to Libai
+
+`oneflow.nn.Module` implements the same interface as `torch.nn.Module`, so you can simply replace `torch` in the model definition with `oneflow` to embed the existing model structure into Libai.
```
Comment (rentainhe) [inline docs/source/tutorials/Import Pytorch Models.md:3]: `oneflow.nn.Module` implements the same interface as `torch.nn.Module`, so it's easy for users to convert a pytorch model to oneflow model and train it based on LiBai

这样会不会好些

File: docs/source/tutorials/Import Pytorch Models.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
+# Import PyTorch models to Libai
```
Comment (rentainhe) [inline docs/source/tutorials/Import Pytorch Models.md:1]: 我感觉标题可以改一下，因为这个部分其实有点像介绍`import oneflow as torch`，感觉`Import Pytorch models to LiBai`有点奇怪，其他人有没有什么想法

Comment (Ldpe2G) [reply docs/source/tutorials/Import Pytorch Models.md:1]: Convert PyTorch Models to Libai，这样呢

Comment (rentainhe) [reply docs/source/tutorials/Import Pytorch Models.md:1]: 感觉这样也行，但是模型得挑一个有说服力的，新的，建议是ConvNeXt，然后把它的训练超参对齐，列举一个完整的config

File: docs/source/tutorials/Import Pytorch Models.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+`oneflow.nn.Module` implements the same interface as `torch.nn.Module`, so you can simply replace `torch` in the model definition with `oneflow` to embed the existing model structure into Libai.
+
+Take [MobileNetV2](https://github.com/d-li14/mobilenetv2.pytorch/blob/master/models/imagenet/mobilenetv2.py) training cifar100 as an example.
+
+### Replace `torch` with `oneflow`
```
Comment (Ldpe2G) [inline docs/source/tutorials/Import Pytorch Models.md:7]: Steps for converting `torch` model script to `oneflow` ?

File: docs/source/tutorials/Import Pytorch Models.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+libai/model/MobileNetV2.py (original code)
```
Comment (Ldpe2G) [inline docs/source/tutorials/Import Pytorch Models.md:9]: Take MobileNetV2 as an example, the original implementation is simplified as follows:

这样好些

File: docs/source/tutorials/Import Pytorch Models.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+... # omit some code
+
+def mobilenetv2(**kwargs):
+    """
+    Constructs a MobileNet V2 model
+    """
+    return MobileNetV2(**kwargs)
+```
+
+It is only necessary to **modify the `torch` in the import statement to `oneflow`** to convert it into an oneflow model while ensuring the model structure is identical. Then add the training loss calculation at the end of the model to form a complete training model, i.e.
```
Comment (Ldpe2G) [inline docs/source/tutorials/Import Pytorch Models.md:34]: To convert the above script into oneflow, the only thing user need to do is simply changing `import torch.nn as nn` to `import oneflow.nn as nn`. Then add then loss module at the end of the forward function to form a complete training module.

这样如何

File: docs/source/tutorials/Import Pytorch Models.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+libai/model/MobileNetV2.py
```
Comment (Ldpe2G) [inline docs/source/tutorials/Import Pytorch Models.md:36]: 这个路径是想表明，用户增的模型需要放到 `libai/model` 路径下么，目前没有这个限制吧？

File: docs/source/tutorials/Import Pytorch Models.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        if labels is not None and self.training:
+            loss = self.loss(logits, labels.long())
+            return {"loss": loss}
+        else:
+            return {"prediction_scores": logits}
+```
+
+Now we have obtained a MobileNetV2 model that is usable under Libai.
+
+### Set the training configuration
```
Comment (Ldpe2G) [inline docs/source/tutorials/Import Pytorch Models.md:77]: Create configuration file

File: docs/source/tutorials/Import Pytorch Models.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+Create a configuration file: configs/cifar.py
```
Comment (Ldpe2G) [inline docs/source/tutorials/Import Pytorch Models.md:79]: For example `configs/cifar.py`

File: docs/source/tutorials/Import Pytorch Models.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+from .common.models.graph import graph
+from .common.train import train
+from .common.optim import optim
+from .common.data.cifar100 import dataloader
+
+model = LazyCall(MobileNetV2Training)()
+```
+
+### Modify the configuration of the cifar100 dataset
+configs/common/data/cifar100.py (modify the `mixup_func` in dataloader.train to `None`)
```
Comment (Ldpe2G) [inline docs/source/tutorials/Import Pytorch Models.md:93]: 直接修改这个 cifar100.py 么，应该是在 上面那个 config 文件里 修改 `dataloader.train.mixup_func=None` 吧

File: docs/source/tutorials/Import Pytorch Models.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+        ),
+    ],
+    num_workers=12,
+    mixup_func=None
+)
+```
+
+### Start training
+
+The imported PyTorch model structure can be used for training by running `zsh tools/train.sh tools/train_net.py configs/cifar.py 1` in the Libai directory.
```
Comment (Ldpe2G) [inline docs/source/tutorials/Import Pytorch Models.md:113]: After the above preparation, users can start training on single GPU by running `bash tools/train.sh tools/train_net.py configs/cifar.py 1` under Libai.

这样如何

File: docs/source/tutorials/Import Pytorch Models.md (hunk 1 above)
Comment (marigoold) [reply docs/source/tutorials/Import Pytorch Models.md:3]: 嗯，我觉得这样更书面一些，我这里写得有点口语化了

File: docs/source/tutorials/Import Pytorch Models.md (hunk 2 above)
Comment (marigoold) [reply docs/source/tutorials/Import Pytorch Models.md:1]: 我个人感觉convert有些让用户迁移到Libai的感觉，但是这个tutorial的本意是给用户示例如何在Libai中用PyTorch的模型。不过convert看起来确实更常见一些，我改掉吧

File: docs/source/tutorials/Import Pytorch Models.md (hunk 6 above)
Comment (marigoold) [reply docs/source/tutorials/Import Pytorch Models.md:36]: 那这个路径应该放在哪里呢？放在projects里面吗？

File: docs/source/tutorials/Import Pytorch Models.md (hunk 9 above)
Comment (marigoold) [reply docs/source/tutorials/Import Pytorch Models.md:93]: 我傻了，这里确实改config就好

File: docs/source/tutorials/Import Pytorch Models.md (hunk 10 above)
Comment (marigoold) [reply docs/source/tutorials/Import Pytorch Models.md:113]: 我觉得你的描述更好一些

File: docs/source/tutorials/Import Pytorch Models.md (hunk 2 above)
Comment (L1aoXingyu) [reply docs/source/tutorials/Import Pytorch Models.md:1]: Train PyTorch models with LiBai seamlessly
这样呢，我希望这个 tutorial 是教会用户如何把自己的模型放到 libai 下来训练，只需要做一个最小化修改

File: docs/source/tutorials/Import Pytorch Models.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+Take [MobileNetV2](https://github.com/d-li14/mobilenetv2.pytorch/blob/master/models/imagenet/mobilenetv2.py) training cifar100 as an example.
+
+### Replace `torch` with `oneflow`
+
+libai/model/MobileNetV2.py (original code)
+
+```python
+"""
+Creates a MobileNetV2 Model as defined in:
```
Comment (L1aoXingyu) [inline docs/source/tutorials/Import Pytorch Models.md:13]: 这个部分可以删掉

File: docs/source/tutorials/Import Pytorch Models.md (hunk 6 above)
Comment (L1aoXingyu) [reply docs/source/tutorials/Import Pytorch Models.md:36]: 路径可以任意在哪里都可以，用户可以使用 pip install libai，然后只需要2-3个 python 文件就在训练了

File: docs/source/tutorials/Import Pytorch Models.md (hunk 11 above)
Comment (L1aoXingyu) [reply docs/source/tutorials/Import Pytorch Models.md:13]: 这里甚至可以给一个模型的网站，不用把模型代码写在下面

File: docs/source/tutorials/Import Pytorch Models.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+    def forward(self, images, labels):
+        logits = self.model(images)
+        if labels is not None and self.training:
+            loss = self.loss(logits, labels.long())
+            return {"loss": loss}
+        else:
+            return {"prediction_scores": logits}
+```
+
+Now we have obtained a MobileNetV2 model that is usable under Libai.
```
[the remark this answers is not in the document]
Comment (marigoold) [reply docs/source/tutorials/Import Pytorch Models.md:75]: 这么说来我感觉这句话后面几个介词也有问题，中式英语味太重了，我改一下措辞

File: docs/source/tutorials/Import Pytorch Models.md (hunk 3 above)
Comment (marigoold) [reply docs/source/tutorials/Import Pytorch Models.md:7]: 这里就做了一个操作，把torch替换成oneflow，用steps是不是有些不准？

File: docs/source/tutorials/Import Pytorch Models.md (hunk 1 above)
Comment (marigoold) [reply docs/source/tutorials/Import Pytorch Models.md]: 已修改

File: docs/source/tutorials/Import Pytorch Models.md (hunk 2 above)
Comment (marigoold) [reply docs/source/tutorials/Import Pytorch Models.md]: 已修改

File: docs/source/tutorials/Import Pytorch Models.md (hunk 4 above)
Comment (marigoold) [reply docs/source/tutorials/Import Pytorch Models.md]: 已修改

File: docs/source/tutorials/Import Pytorch Models.md (hunk 5 above)
Comment (marigoold) [reply docs/source/tutorials/Import Pytorch Models.md]: 已修改

File: docs/source/tutorials/Import Pytorch Models.md (hunk 8 above)
Comment (marigoold) [reply docs/source/tutorials/Import Pytorch Models.md]: 已修改

File: docs/source/tutorials/Import Pytorch Models.md (hunk 9 above)
Comment (marigoold) [reply docs/source/tutorials/Import Pytorch Models.md]: 已修改
